pkg/infra/x/context/user.go

```diff
@@ -0,0 +1,98 @@
+package store
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/contexthandler/ctxkey"
+	grpccontext "github.com/grafana/grafana/pkg/services/grpcserver/context"
+	"github.com/grafana/grafana/pkg/services/user"
+)
+
+type ctxUserKey struct{}
```

Comment:
maybe extract lines 15-41 to a separate PR to close https://github.com/grafana/grafana/issues/50983?

Reply:
👍 will do